File: src/main/java/wraith/fabricaeexnihilo/modules/ModTools.java
package wraith.fabricaeexnihilo.modules;

import net.fabricmc.fabric.api.item.v1.FabricItemSettings;
import net.minecraft.item.Item;
import net.minecraft.item.ToolMaterials;
import net.minecraft.util.Identifier;
import net.minecraft.util.registry.Registry;
import wraith.fabricaeexnihilo.FabricaeExNihilo;
import wraith.fabricaeexnihilo.modules.materials.ModToolMaterials;
import wraith.fabricaeexnihilo.modules.tools.CrookItem;
import wraith.fabricaeexnihilo.modules.tools.HammerItem;

import java.util.HashMap;
import java.util.Locale;
import java.util.Map;

import static wraith.fabricaeexnihilo.FabricaeExNihilo.id;

// TODO: merge with ModItems (why is this a thing?)
public class ModTools {

    public static final Map<Identifier, Item> CROOKS = new HashMap<>();
    public static final Map<Identifier, Item> HAMMERS = new HashMap<>();

    static {
        for (var toolMaterial : ModToolMaterials.values()) {
            var crookId = id(toolMaterial.name().toLowerCase(Locale.ROOT) + "_crook");
            var crook = new CrookItem(toolMaterial, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP));
            CROOKS.put(crookId, crook);
        }
        HAMMERS.put(id("wooden_hammer"), new HammerItem(ToolMaterials.WOOD, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
        HAMMERS.put(id("stone_hammer"), new HammerItem(ToolMaterials.STONE, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
        HAMMERS.put(id("iron_hammer"), new HammerItem(ToolMaterials.IRON, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
        HAMMERS.put(id("golden_hammer"), new HammerItem(ToolMaterials.GOLD, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
        HAMMERS.put(id("diamond_hammer"), new HammerItem(ToolMaterials.DIAMOND, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
        HAMMERS.put(id("netherite_hammer"), new HammerItem(ToolMaterials.NETHERITE, new FabricItemSettings().group(FabricaeExNihilo.ITEM_GROUP)));
    }

    public static void registerItems() {
        CROOKS.forEach((identifier, item) -> Registry.register(Registry.ITEM, identifier, item));
        HAMMERS.forEach((identifier, item) -> Registry.register(Registry.ITEM, identifier, item));
    }

}

File: src/main/kotlin/exnihilofabrico/modules/ModTools.kt
package exnihilofabrico.modules

import exnihilofabrico.ExNihiloFabrico
import exnihilofabrico.id
import exnihilofabrico.modules.materials.ModToolMaterials
import exnihilofabrico.modules.tools.CrookTool
import exnihilofabrico.modules.tools.HammerTool
import net.minecraft.item.Item
import net.minecraft.item.ToolMaterials
import net.minecraft.util.Identifier
import net.minecraft.util.registry.Registry

object ModTools {
    val tool_settings: Item.Settings = Item.Settings().group(ExNihiloFabrico.ITEM_GROUP).maxCount(1)

    val CROOKS = ModToolMaterials.values().map {
        id("crook_${it.name.toLowerCase()}") to
                CrookTool(it, Item.Settings().group(ExNihiloFabrico.ITEM_GROUP).maxCount(1).maxDamage(it.durability))
    }.toMap()

    val HAMMERS = mapOf<Identifier, Item>(
        id("hammer_wood") to HammerTool(ToolMaterials.WOOD, tool_settings),
        id("hammer_stone") to HammerTool(ToolMaterials.STONE, tool_settings),
        id("hammer_iron") to HammerTool(ToolMaterials.IRON, tool_settings),
        id("hammer_gold") to HammerTool(ToolMaterials.GOLD, tool_settings),
        id("hammer_diamond") to HammerTool(ToolMaterials.DIAMOND, tool_settings)
    )

    fun registerItems(registry: Registry<Item>) {
        CROOKS.forEach { (k, v) -> Registry.register(registry, k, v) }
        HAMMERS.forEach { (k, v) -> Registry.register(registry, k, v) }
    }
}